tests/test_koji_util.py

```diff
@@ -101,16 +105,57 @@ def test_create_simple_session(self):
 
         (flexmock(koji_util.koji).should_receive('ClientSession').with_args(
             url, opts={'krb_rdns': False}).and_return(session))
-        assert create_koji_session(url) == session
+        assert create_koji_session(url)._wrapped_session == session
 
-    def test_create_authenticated_session(self):
+    @pytest.mark.parametrize(('ssl_session'), [
+        (True, False),
+    ])
+    def test_create_authenticated_session(self, tmpdir, ssl_session):
         url = 'https://koji-hub-url.com'
+        args = {}
+
         session = flexmock()
-        session.should_receive('krb_login').once().and_return(True)
+        if ssl_session:
+            args['ssl_certs_dir'] = str(tmpdir)
+            session.should_receive('ssl_login').and_return(True)
+        else:
+            session.should_receive('krb_login').once().and_return(True)
 
         (flexmock(koji_util.koji).should_receive('ClientSession').with_args(
             url, opts={'krb_rdns': False}).and_return(session))
-        assert create_koji_session(url, {}) == session
+        assert create_koji_session(url, args)._wrapped_session == session
+
+    @pytest.mark.parametrize(('session_type', 'session_args'), [
+        ('ssl_login', {'ssl_certs_dir': 'value'}),
+        ('krb_login', {}),
+    ])
+    @pytest.mark.parametrize(('error_type', 'should_recover', 'attempts'), [
+        (ConnectionError, True, HTTP_MAX_RETRIES - 1),
+        (ConnectionError, False, HTTP_MAX_RETRIES),
+        (KeyError, False, 1),
+    ])
+    def test_create_session_failures(self, tmpdir, session_type, session_args,
+                                     error_type, should_recover, attempts):
+        url = 'https://koji-hub-url.com'
```

Comment:
Let's always use `.example.com` for domains in test cases.

Reply:
Fixed.